backend/src/test/java/com/github/esgoet/backend/services/BookServiceUnitTest.java

```diff
@@ -96,4 +97,52 @@ void deleteBook_Test() {
         bookService.deleteBook("1");
         verify(bookRepo).deleteById("1");
     }
+
+    @Test
+    void testUpdateBook_Success() {
+
+        // Given
+        String id = "1";
+        Book existingBook = new Book(id, "author1", "title1", Genre.THRILLER, "description1", "12345678", "cover1", localDate);
+        BookDto updatedBook = new BookDto("author2", "title2", Genre.FICTION, "description2", "23456789", "cover2", localDate);
+
+        // When
+        when(bookRepo.findById(id)).thenReturn(Optional.of(existingBook));
+        when(bookRepo.save(any(Book.class))).thenAnswer(invocation -> invocation.getArguments()[0]);
+
+        Book result = bookService.updateBook(updatedBook, id);
+
+        // Then
+        assertNotNull(result);
+        assertEquals(updatedBook.title(), result.title());
+        assertEquals(updatedBook.author(), result.author());
+        assertEquals(updatedBook.genre(), result.genre());
+        assertEquals(updatedBook.description(), result.description());
+        assertEquals(updatedBook.cover(), result.cover());
+        assertEquals(updatedBook.isbn(), result.isbn());
+        assertEquals(updatedBook.publicationDate(), result.publicationDate());
+        verify(bookRepo).findById(id);
+        verify(bookRepo).save(any(Book.class));
+    }
+
+    @Test
+    void testUpdateBook_BookNotFound() {
+
+        // Given
+        String id = "1";
+        BookDto updatedBook = new BookDto("author", "title", Genre.FICTION, "description", "isbn", "cover", localDate);
```

Comment:
would again do the same, but here leaving it as it is isn't so complicated as before
```suggestion
        BookDto updatedBookDto= new BookDto("author", "title", Genre.FICTION, "description", "isbn", "cover", localDate);
        Book updatedBook = new Book(id, "author", "title", Genre.FICTION, "description", "isbn", "cover", localDate);
```